drivers/azure/azureutil/azureutil.go

```diff
@@ -232,21 +232,31 @@ func (a AzureClient) GetSubnet(resourceGroup, virtualNetwork, name string) (netw
 	return a.subnetsClient().Get(resourceGroup, virtualNetwork, name, "")
 }
 
+// CreateSubnet creates or updates a subnet if it does not already exist.
 func (a AzureClient) CreateSubnet(ctx *DeploymentContext, resourceGroup, virtualNetwork, name, subnetPrefix string) error {
+	subnet, err := a.subnetsClient().Get(resourceGroup, virtualNetwork, name, "")
+	if subnet.Name != nil {
```

Comment:
I would rather handle this using the `err` value. 
- If `err` is nil, that means the subnet exists and we can exit. 
- If it does not, we should cast it to the `autorest.DetailedError` type. 
  - If `err.StatusCode` is `http.NotFound`, we can proceed with creating the subnet; 
  - if it's something else, we should probably log and abort.

Reply:
Hm, I thought that ```err``` is only providing information but connection problems and so on, but well I will have a look,